### DISCUSSION TITLE
GUI is not showing

### DISCUSSION THREAD

**Discussion by SilentException** [timmo001/system-bridge#1933] [Help]
### Description

After clicking on tray icon and opening Settings or View data, empty System Bridge window appears with errors in backend console. Image and logs attached.

[system-bridge-gui (settings).log](https://github.com/timmo001/system-bridge/files/8939480/system-bridge-gui.settings.log)
[system-bridge (settings).log](https://github.com/timmo001/system-bridge/files/8939478/system-bridge.settings.log)
[system-bridge-console (settings).log](https://github.com/timmo001/system-bridge/files/8939479/system-bridge-console.settings.log)
![system-bridge-gui (setting)](https://user-images.githubusercontent.com/613606/174576153-43ef24bc-cbb8-4f6d-b976-af6c388abbb2.jpg)
.

### What Platform / OS are you running?

Windows

### What version are you running?

3.2.1

### Anything in the logs or a references that might be useful?

```txt
Besides the main issue I can see that there is also errors related to database and SQL transactions in backend console log.
```

**Comment by timmo001** [comment]
Please upgrade and see if your issues are fixed

**Comment by SilentException** [comment]
Hi @timmo001 , I'm afraid I don't really see much improvement.

First I removed data folder to start fresh. (c:\Users\Mitja\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0\LocalCache\Local\timmo001)

Still some SQL errors on startup and the GUI is still blank. I clicked on "show settings" on 14:53:14 timestamp. Everything before that was simply me running python command and watching the output (no input from my side).

[system-bridge.logs.zip](https://github.com/timmo001/system-bridge/files/9054926/system-bridge.logs.zip)

**Comment by kjake** [comment]
I also have this problem.  Additionally, I get an error of:
> The code execution cannot proceed because Qt6WebEngineCore.dIl was not found. Reinstalling the program may fix this problem.

`pyside6` is installed without problem, confirmed by running `python -m pip install --upgrade pyside6`

System Bridge 3.4.1 is installed.

I can open the GUI log, grab the URL, and open it in a browser though.

**Comment by kjake** [comment]
FYI, same occurs with 3.4.4

**Comment by timmo001** [comment]
I had an issue with a laptop that had pyside6 installed but not fully. I ended up running:

```bash
python -m pip install --force --upgrade PySide6
````

which fixed the issue of PySide7 not being found

**Comment by kjake** [comment]
I figured out my problem.

1. Don't use Python from the Windows Store - it is a bad idea and not recommended by the Python devs.
2. Install Python directly from the Python website, and ensure to check the checkbox which says "Add Python [VER] to PATH" - this is critical for Python to work correctly from the CLI and for third-party dependent libraries to load, like Qt6WebEngineCore.dIl.  @timmo001 - I would recommend stressing this in your documentation (Windows Store does not setup PATH and does not give you an option to).

![CleanShot 2022-08-16 at 20 47 17](https://user-images.githubusercontent.com/1299894/185010835-5dbc3ea3-f8d8-4f7c-948f-24ceba092828.jpg)

**Comment by SilentException** [comment]
I also use Python from Windows Store. Solved missing Qt6WebEngineCore.dll by adding new OS environment variable 

`QTWEBENGINEPROCESS_PATH=%USERPROFILE%\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0\LocalCache\local-packages\Python310\site-packages\PySide6\`

But the GUI from within the app is still not shown. It shows the main window, it blinks once and then this is the logged error. I can still access the GUI from the browser but would be nice to have this working as well.

```
2022-08-17 13:26:36 INFO (MainThread) [systembridgegui.window.main.MainWindow] Open URL: PySide6.QtCore.QUrl('http://localhost:9170/app/settings.html?apiKey=d73c9a19-4069-43f2-add1-1e972839dc32&apiPort=9170')
qt.webenginecontext:

GL Type: desktop
Surface Type: OpenGL
Surface Profile: CompatibilityProfile
Surface Version: 4.3
QSG RHI Backend: OpenGL
Using Supported QSG Backend: yes
Using Software Dynamic GL: no
Using Multithreaded OpenGL: yes

Init Parameters:
  *  application-name python3
  *  browser-subprocess-path C:\Users\Mitja\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0\LocalCache\local-packages\Python310\site-packages\PySide6\
  *  create-default-gl-context
  *  disable-es3-gl-context
  *  disable-features ConsolidatedMovementXY,InstalledApp,BackgroundFetch,WebOTP,WebPayments,WebUSB,PictureInPicture
  *  disable-speech-api
  *  enable-features NetworkServiceInProcess,TracingServiceInProcess
  *  enable-threaded-compositing
  *  in-process-gpu
  *  use-gl desktop

QOpenGLShader::compile(Fragment): ERROR: 4:1: 'textureSampler' : syntax error syntax error


*** Problematic Fragment shader source code ***
#version 110
#ifdef GL_KHR_blend_equation_advanced
#extension GL_ARB_fragment_coord_conventions : enable
#extension GL_KHR_blend_equation_advanced : enable
#endif
#define lowp
#define mediump
#define highp
#line 1
varying highp vec2 uv;uniform sampler2DRect textureSampler;uniform bool swizzle;uniform highp float opacity;void main() {   highp vec4 tmpFragColor = texture2DRect(textureSampler,uv);   tmpFragColor.a *= opacity;   gl_FragColor = swizzle ? tmpFragColor.bgra : tmpFragColor;}
***
```

This is Windows 10 headless server and I tried from RDP session as well as from TeamViewer (thinking that display driver is to blame).

**Comment by kjake** [comment]
I've done all of my testing via Remote Desktop.  As I mentioned in my previous posting, do not use Python from Windows Store - it's part of what caused my problems.

Here's some more reading on the weirdness of fake pathing that the Store does to its packages:
[https://github.com/python/cpython/blob/main/Doc/using/windows.rst#the-microsoft-store-package](https://github.com/python/cpython/blob/main/Doc/using/windows.rst#the-microsoft-store-package)
